### ISSUE TITLE
Any methods to use a pytorch model to generate trajectory in C++ side?

### ISSUE THREAD

**Issue by 87a** [deepmind/open_spiel#928]
Hi there, suppose I have a pytorch model, and I want to use it to generate trajectories for reinforcement learning, for example REINFORCE. The code in python should be like
```
state = game.new_initial_state()
while not state.is_terminal():
    action = use_model_to_get_action(state)
    state.apply_action(action)
    save_data() # like state, action, reward, next_state
```
but it seems like a bit slow in python, so i'm wondering if I can use pybind and libtorch to generate trajectories in parallel and get buffers return to python. But I can't find an example, that is, is it possible to pass a network to libtorch and get tensors which need gradients back to python?

Also, I don't know whether it will be faster or should I write this totally in C++?

Any idea for this?

**Comment by lanctot** [comment]
We do not have anything like this, but it would be quite useful to have.

We have a few things, though, that are quite relevant in writing this:

- There is an old (unused) version of what you're describing in Tensorflow, see here: https://github.com/deepmind/open_spiel/tree/master/open_spiel/contrib
- There is an implementation of AlphaZero based entirely on C++ LibTorch, see here: https://github.com/deepmind/open_spiel/tree/master/open_spiel/algorithms/alpha_zero_torch
- There is an implementation of DQN based entirely on C++ LibTorch, see here: https://github.com/deepmind/open_spiel/tree/master/open_spiel/algorithms/dqn_torch

Something like this would make a very welcome contribution to OpenSpiel as it could benefit the community greatly.

**Comment by lanctot** [comment]
Closing due to inactivity. Please re-open if you want to continue the discussion.
